internal/provider/user_role_grant_resource_test.go

```diff
@@ -171,61 +198,80 @@ func TestIntegrationRoleGrantResource(t *testing.T) {
+
 	// Called by the Import step
 	s.EXPECT().GetAllRolesForUser(gomock.Any(), userID).
-		Return(&PostCreateGetResponse, nil, nil).Times(1)
+		Return(&PostDeleteGetResponse, nil, nil)
 
 	// Called by Delete
+	s.EXPECT().GetServiceAccount(gomock.Any(), sa.Id).
+		Return(&sa, &http.Response{Status: http.StatusText(http.StatusOK)}, nil)
+
+	s.EXPECT().GetCluster(gomock.Any(), clusterID).
+		Return(&cluster, &http.Response{Status: http.StatusText(http.StatusOK)}, nil)
+
+	s.EXPECT().GetAllRolesForUser(gomock.Any(), userID).
+		Return(&PostDeleteGetResponse, nil, nil).Times(2)
+
 	s.EXPECT().RemoveUserFromRole(gomock.Any(), userID, string(client.RESOURCETYPETYPE_CLUSTER), clusterID, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER)).
 		Return(nil, nil, nil)
-	s.EXPECT().RemoveUserFromRole(gomock.Any(), userID, string(client.RESOURCETYPETYPE_ORGANIZATION), "", string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR)).
+
+	s.EXPECT().DeleteServiceAccount(gomock.Any(), gomock.Any()).
 		Return(nil, nil, nil)
+
 	s.EXPECT().DeleteCluster(gomock.Any(), gomock.Any()).
 		Return(nil, nil, nil)
 
-	// Post cleanup verification
-	s.EXPECT().GetAllRolesForUser(gomock.Any(), userID).
-		Return(&PreCreateGetResponse, nil, nil)
-
-	testRoleGrantResource(t, clusterName, false /* skipOrgTest */, true /* useMock */)
+	testRoleGrantResource(t, clusterName, true /* useMock */)
 }
 
-func testRoleGrantResource(t *testing.T, clusterName string, skipOrgTest, useMock bool) {
+func testRoleGrantResource(t *testing.T, clusterName string, useMock bool) {
 	var (
 		orgGrantResourceName     = "cockroach_user_role_grant.org_grant"
 		clusterGrantResourceName = "cockroach_user_role_grant.cluster_grant"
-		clusterResourceName      = "cockroach_cluster.test"
 	)
 	defer HookGlobal(&generateRandomPassword, func() (string, error) {
 		return testPassword, nil
 	})()
 
-	// compose checkFuncs so we can optionally exclude the org test.
 	checkFuncs := []resource.TestCheckFunc{
-		testRoleGrantExists(clusterGrantResourceName, clusterResourceName, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER), string(client.FOLDERRESOURCETYPETYPE_CLUSTER)),
-	}
-	if !skipOrgTest {
-		checkFuncs = append(checkFuncs, testRoleGrantExists(orgGrantResourceName, clusterResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION)))
+		testRoleGrantExists(clusterGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_CLUSTER_OPERATOR_WRITER), string(client.FOLDERRESOURCETYPETYPE_CLUSTER)),
+		testRoleGrantExists(orgGrantResourceName, string(client.ORGANIZATIONUSERROLETYPE_BILLING_COORDINATOR), string(client.RESOURCETYPETYPE_ORGANIZATION)),
 	}
 
 	resource.Test(t, resource.TestCase{
-		// After removal of the test resources, verify that the user running the
-		// tests still has at least one non ORG_MEMBER role to show that this
-		// deletion did not remove the other roles like the user_role_grants
```

Comment:
instead of using CheckDestroy, I added a third step which removes one of the resources